src/drivers/differential_pressure/asp5033driver/ASP5033Driver.cpp

```diff
@@ -0,0 +1,240 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020-2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+
+
+#include "ASP5033Driver.hpp"
+
+ASP5033Driver::ASP5033Driver(const I2CSPIDriverConfig &config) :
+	I2C(config),
+	I2CSPIDriver(config)
+
+{
+	_differential_pressure_pub.advertise();
+
+}
+
+ASP5033Driver::~ASP5033Driver()
+{
+	perf_free(_sample_perf);
+	perf_free(_comms_errors);
+	perf_free(_fault_perf);
+}
+
+int ASP5033Driver::probe()
+{
+	uint8_t cmd=REG_CMD_ASP5033;
+	int ret = transfer(&cmd, 1, nullptr, 0);
+	return ret;
+
+}
+
+int ASP5033Driver::init()
+{
+	int ret = I2C::init();
+	if (ret != PX4_OK) {
+		DEVICE_DEBUG("I2C::init failed (%i)", ret);
+		return ret;
+	}
+
+	if (ret == PX4_OK) {
+		DEVICE_DEBUG("I2C::init successed (%i)", ret);
+		ScheduleNow();
+	}
+
+	return ret;
+}
+
+
+int ASP5033Driver::get_differential_pressure()
+{
+	if(((double)(clock()-last_sample_time)/CLOCKS_PER_SEC)>0.1){
+		return 0;
+	}
```

Comment:
We don't usually use this one but instead `hrt_absolute_time()` which gives you a monotonic timestamp in microseconds.

And then here you can use `hrt_elapsed_time()`.